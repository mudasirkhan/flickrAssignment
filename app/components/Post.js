import React, { PureComponent } from 'react';
import styles from "../styles/styles";
import {Image, TouchableOpacity} from 'react-native';
export default class Post extends PureComponent {
    constructor(props) {
        super(props);
    }
    render() {
        const {item, zoom, index, onPress} = this.props;
        return  <TouchableOpacity
            activeOpacity={1}
            onPress={()=>{
                onPress(index)
                // this.props.navigation.navigate('Image',{uri:`http://farm${item.farm}.static.flickr.com/${item.server}/${item.id}_${item.secret}.jpg`})
            }}>
            <Image
                defaultSource={require('../../assets/splash.png') }
                source={{uri:`http://farm${item.farm}.static.flickr.com/${item.server}/${item.id}_${item.secret}.jpg`,}}
                style={zoom ? styles.listItemLarge :styles.listItemSmall}
                resizeMode={"cover"}/>
        </TouchableOpacity>
    }
}