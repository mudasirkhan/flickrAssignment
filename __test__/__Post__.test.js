// Link.react.test.js
import React from 'react';
import Post from '../app/components/Post';
import renderer from 'react-test-renderer';

function onPress() {
    console.log("working");
}

test('Link changes the class when hovered', () => {
    const component = renderer.create(
        <Post
            item={{
                farm: 2,
                id: "44867966125",
                isfamily: 0,
                isfriend: 0,
                ispublic: 1,
                owner: "119451656@N06",
                secret: "d9c9383ace",
                server: "1979",
                title: "Campeonato Estadual Laser 2018-1977.jpg"
            }}
            zoom={true}
            onPress={onPress}
            index={0}
        />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    //tree.props.onPress();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});