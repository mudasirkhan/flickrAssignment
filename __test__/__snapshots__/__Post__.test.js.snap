// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignSelf": "flex-end",
          "backgroundColor": "#5b5b5b",
          "borderRadius": 16,
          "opacity": 0.75,
          "padding": 16,
          "position": "absolute",
          "width": 120,
          "zIndex": 99,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
          }
        }
      >
        Tap to close
      </Text>
    </View>
    <Image
      defaultSource={
        Object {
          "testUri": "../../../assets/splash.png",
        }
      }
      resizeMode="cover"
      source={
        Object {
          "uri": "http://farm2.static.flickr.com/1979/44867966125_d9c9383ace.jpg",
        }
      }
      style={
        Object {
          "height": 750,
          "width": 750,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "padding": 8,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "maxWidth": 200,
          }
        }
      >
        Title: 
        Campeonato Estadual Laser 2018-1977.jpg
      </Text>
      <Text>
        Farm: 
        2
      </Text>
    </View>
  </View>
</View>
`;

exports[`Link changes the class when hovered 2`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignSelf": "flex-end",
          "backgroundColor": "#5b5b5b",
          "borderRadius": 16,
          "opacity": 0.75,
          "padding": 16,
          "position": "absolute",
          "width": 120,
          "zIndex": 99,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
          }
        }
      >
        Tap to close
      </Text>
    </View>
    <Image
      defaultSource={
        Object {
          "testUri": "../../../assets/splash.png",
        }
      }
      resizeMode="cover"
      source={
        Object {
          "uri": "http://farm2.static.flickr.com/1979/44867966125_d9c9383ace.jpg",
        }
      }
      style={
        Object {
          "height": 750,
          "width": 750,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "padding": 8,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "maxWidth": 200,
          }
        }
      >
        Title: 
        Campeonato Estadual Laser 2018-1977.jpg
      </Text>
      <Text>
        Farm: 
        2
      </Text>
    </View>
  </View>
</View>
`;

exports[`Link changes the class when hovered 3`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignSelf": "flex-end",
          "backgroundColor": "#5b5b5b",
          "borderRadius": 16,
          "opacity": 0.75,
          "padding": 16,
          "position": "absolute",
          "width": 120,
          "zIndex": 99,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
          }
        }
      >
        Tap to close
      </Text>
    </View>
    <Image
      defaultSource={
        Object {
          "testUri": "../../../assets/splash.png",
        }
      }
      resizeMode="cover"
      source={
        Object {
          "uri": "http://farm2.static.flickr.com/1979/44867966125_d9c9383ace.jpg",
        }
      }
      style={
        Object {
          "height": 750,
          "width": 750,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "padding": 8,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "maxWidth": 200,
          }
        }
      >
        Title: 
        Campeonato Estadual Laser 2018-1977.jpg
      </Text>
      <Text>
        Farm: 
        2
      </Text>
    </View>
  </View>
</View>
`;
